enterprise/views.py

```diff
@@ -479,45 +480,69 @@ def get_enterprise_course_enrollment_page(self, request, enterprise_uuid, course
             raise Http404
 
         try:
+            enrollment_client = EnrollmentApiClient()
+            modes = enrollment_client.get_course_modes(course_id)
+        except HttpClientError:
+            logger.error('Failed to determine available course modes for course ID: %s', course_id)
+            raise Http404
+
+        try:
             enterprise_uuid = UUID(enterprise_uuid)
             # pylint: disable=no-member
             enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_uuid)
         except (ValueError, EnterpriseCustomer.DoesNotExist):
             logger.error('Unable to find enterprise customer for UUID: %s', enterprise_uuid)
             raise Http404
 
+        course_modes = []
+
+        audit_modes = getattr(settings, 'ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES', ['audit'])
+
+        for mode in modes:
+            if mode['min_price']:
+                price_text = '${}'.format(mode['min_price'])
+            else:
+                price_text = self.context_data['free_price_text']
+            if mode['slug'] in audit_modes:
+                description = self.context_data['audit_text']
+            else:
+                description = self.context_data['verified_text']
+            course_modes.append({
+                'mode': mode['slug'],
+                'title': mode['name'],
+                'original_price': price_text,
+                'final_price': price_text,
```

Comment:
@haikuginger @douglashall my understanding on showing final price was to show the discounted price after applying the related enterprise coupon (if available).

For this we have to:
1. verify if the current course exists in enterprise catalog
2. get enterprise related coupon (enterprise entitlement)
3. verify if the coupon is valid for current user
4. get the final price after applying the coupon

The above process is currently implemented in Ecom in the method [get_entitlement_voucher](https://github.com/edx/ecommerce/blob/master/ecommerce/enterprise/entitlements.py#L30) and we will need same procedure for showing the final price. So that the learner view the exact discounted price on enterprise landing page as available on the basket checkout page in Ecom for enterprise auto entitlements.